### PR TITLE
Add linter CI for YAML files

### PR DESCRIPTION
**Pull request by fmuyassarov**
Currently we don't lint YAML files. As a result, sometimes we see
issues with some YAML code being merged and the issue is identified
later on. Example, trailing spaces which might breake the
configuration file of Prow.

This patch adds yamllinter based GitHub action so that we verify
correctness of YAML files on every pull request submission, before
landing on the main branch. As a start, trailing space, key-duplicate
and truthy rules are enabled, and the list can be extended based on
the need.
Fixes: https://github.com/metal3-io/project-infra/issues/322

### CODE REVIEW

File: .github/workflows/yamllint.yaml
Comment (macaptain) [inline .github/workflows/yamllint.yaml:19]: Should we use an image which has yamllint already installed? (e.g. https://hub.docker.com/r/cytopia/yamllint). That way we won't have to install it every time (once the image has been pulled into Prow).

Comment (fmuyassarov) [reply .github/workflows/yamllint.yaml:19]: My initial plan was the same and went this way, because I could not find an official image. But perhaps it is fine to use the one referenced.

Comment (fmuyassarov) [reply .github/workflows/yamllint.yaml:19]: Just heads up, that this is running on GitHub infra.  @macaptain  Do you think it is ok to use https://hub.docker.com/r/cytopia/yamllint ?

Comment (furkatgofurov7) [reply .github/workflows/yamllint.yaml:19]: This looks interesting, we can leave it as is as I found [this](https://github.com/lsst/pipe_base/blob/main/.github/workflows/yamllint.yaml#L18-L19) where it is installed the same way or maybe we could use this [one](https://github.com/frenck/action-yamllint#usage)?

**Review by furkatgofurov7 — COMMENTED** [review]
Looks good, is there any way to test this patch before merging?

**Comment by fmuyassarov** [comment]
> Looks good, is there any way to test this patch before merging?

yep, I have tested it on my work and works fine. Example: https://github.com/fmuyassarov/project-infra/pull/1

File: .github/workflows/yamllint.yaml
Comment (fmuyassarov) [reply .github/workflows/yamllint.yaml:19]: nice, I think using the GA image is the best, I will test it

Comment (fmuyassarov) [reply .github/workflows/yamllint.yaml]: I found `ibiqlik/action-yamllint@v3` image to be more stable and reliable.

**Comment by Rozzii** [comment]
@fmuyassarov  Have you tested this on metal3-dev-env Ansible files? did it work correctly?

**Comment by fmuyassarov** [comment]
> @fmuyassarov Have you tested this on metal3-dev-env Ansible files? did it work correctly?

yes, here is an example https://github.com/fmuyassarov/project-infra/pull/2. Btw, this is not meant to test dev-env scripts (YAML files) for only Prow yaml files in project-infra. Perhaps we can extend it later on, but that's out of scope for now.

File: .github/workflows/yamllint.yaml
Comment (fmuyassarov) [inline .github/workflows/yamllint.yaml:16]: I believe this is leftover. I will remove it now

**Review by lentzi90 — COMMENTED** [review]
Great addition! I would suggest enabling [key-duplicates](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.key_duplicates) and [truthy](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.truthy) also as I think they can prevent some mistakes that are easy to miss otherwise.

File: .github/workflows/yamllint.yaml
Comment (fmuyassarov) [reply .github/workflows/yamllint.yaml]: tested and removed

**Review by namnx228 — APPROVED** [review]
LGTM

**Comment by fmuyassarov** [comment]
> Great addition! I would suggest enabling [key-duplicates](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.key_duplicates) and [truthy](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.truthy) also as I think they can prevent some mistakes that are easy to miss otherwise.

Yep, good suggestion. I will add it now
/hold

**Comment by fmuyassarov** [comment]
@lentzi90 added those two rules.
/hold cancel

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Review by furkatgofurov7 — APPROVED** [review]
Let's try it!

/approve

**Comment by furkatgofurov7** [comment]
Prow is not behaving properly, trying it again.

/lgtm

**Review by kashifest — COMMENTED** [review]
/lgtm

**Pull request merged**
